data_analysis/correction_framework.py

```diff
@@ -0,0 +1,290 @@
+import sys
+import os
+
+import astropy.units as u
+from astroplan import Observer
+from astropy.time import Time
+
+PROJECT_DATE_FORMAT = "%Y-%m-%d %H:%M:%S.%f"
+
+
+class MinuteRecord:
+    def __init__(
+        self,
+        utc: Time,
+        local: Time,
+        temp: float,
+        freq: float,
+        volt: float,
+        id: str,
+    ):
+        self.utc_time = utc
+        self.local_time = local
+        self.temperature = temp
+        self.frequency = freq
+        self.voltag = volt
+        self.sensor_id = id
+
+
+class NightRecord:
+    # create an uninitialized night record
+    def __init__(self):
+        self.night_of = 0
+        self.morning_of = 0
+        self.sunset = 0
+        self.sunrise = 0
+        self.astronomical_twilight = 0
+        self.moon_illumination = 0
+        self.weather = 0
+        self.min_records = []
+        self.valid_for_use = False
+        self.exclusion_reason = "UNINITIALIZED"
+
+    # initialize a night record from a raw data file
+    def initialize(
+        self,
+        night_date: Time,
+        night_file: str,
+        morning_date: Time,
+        morning_file: str,
+        sensor_lat: float,
+        sensor_long: float,
+    ):
```

Comment:
Having an `__init__ ` and `initialize` feels like an anti-pattern., especially since you are doing a good amount of what the `__init__ ` function should be doing in the `initialize` function. Move your variable initializations to the `__init__ ` constructor and change the name of your `initialize` function.

Reply:
Okay, so originally there was only the init function, which read from the raw files of data like the kind uploaded to the Google Drive, but when I made the serialization for the struct, I realized I would need a way to deserialize it back to a struct from its own file, which required a different series of steps than initializing from the raw file. So, I wanted init to just make an empty object that would then be seeded by initialize and rec_import respectively but I didn't know where else to specify the required types of the fields since Python doesn't seem to have strict struct declaration prior to use.